src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -3,33 +3,69 @@
  * @author Box
  */
 import * as React from 'react';
-import flow from 'lodash/flow';
 import { FormattedMessage } from 'react-intl';
+import flow from 'lodash/flow';
+import { MetadataEmptyState } from '@box/metadata-editor';
+import messages from '../common/messages';
 import { withAPIContext } from '../common/api-context';
 import { withErrorBoundary } from '../common/error-boundary';
 import { withLogger } from '../common/logger';
-import { ORIGIN_METADATA_SIDEBAR_REDESIGN } from '../../constants';
-import { EVENT_JS_READY } from '../common/logger/constants';
 import { mark } from '../../utils/performance';
-import messages from '../common/messages';
+import { EVENT_JS_READY } from '../common/logger/constants';
+import API from '../../api';
+import { ORIGIN_METADATA_SIDEBAR_REDESIGN } from '../../constants';
+import type { ErrorContextProps } from '../../common/types/api';
+import type { WithLoggerProps } from '../../common/types/logging';
+import useSidebarMetadataFetcher from './hooks/useSidebarMetadataFetcher';
 
 import './MetadataSidebarRedesign.scss';
 
+export interface ExternalProps {
+    isBoxAiSuggestionsEnabled: boolean;
+    isFeatureEnabled: boolean;
+}
+
+interface PropsWithoutContext extends ExternalProps {
+    elementId: string;
+    fileId: string;
+    hasSidebarInitialized?: boolean;
+}
+
+// For some reason, when changing this type to an interface, the MetadataSidebarRedesign component throws an error on the onError prop.
```

Comment:
Can you share the error? I can see the interface is used in [other PR](https://github.com/box/box-ui-elements/pull/3606/files#diff-eba8562abef4626a4e079dd7ce4779484fc05c0ae63b242451ef3882dc272fa3R48)

Reply:
`Property 'onError' does not exist on type 'MetadataSidebarRedesignProps'.ts(2339)`

I can make the change similarly to another PR where the ErrorContextProps type is defined in the same file. Currently, I'm importing this type from /common/types/api.

<img width="667" alt="Screenshot 2024-08-22 at 14 03 16" src="https://github.com/user-attachments/assets/e100ae4b-5b85-43c7-9f05-8bdf9a9d7e39">

Reply:
`/common/types/api` is using flow types, where `MetadataSidebarRedesign` is using TS - I think that could be the issue. 
Worth to notice, that errors in IDE may sometimes be confusing - it's more reliable to verify console linter checks.

Reply:
Ok i changed that.